packages/decoder/lib/Web3Adapter.ts

```diff
@@ -0,0 +1,50 @@
+import Web3Eth from "web3-eth";
```

Comment:
So, wait, we're not using web3, but we *are* using web3-eth?  Does that mean the problematic parts of web3 are the *rest* of web3, and web3-eth is fine, and we don't have to do anything manually?

If we're OK with using web3-eth, then I'd like to make a suggestion: Ditch the adapter, and just do everything with web3-eth directly in `decoders.ts`.  (Arguably we should be doing that everywhere.)  This adapter doesn't really add anything if that's the case.

(Yeah, this kind of contradicts my previous comment.  I would synthesize them as follows: My primary recommendation would be to ditch the adapter entirely, but *if* you want to keep it, it should live in a separate package, like being part of `InterfaceAdapter`, rather than going here.)